src/wallet/signer.ts

```diff
@@ -0,0 +1,185 @@
+import { BigNumber } from 'bignumber.js'
+import { flatMap } from 'lodash'
+import { decodeAccountID } from 'ripple-address-codec'
+import {
+  decode,
+  encodeForSigning,
+  encodeForSigningClaim,
+} from 'ripple-binary-codec'
+import { sign as signWithKeypair, verify } from 'ripple-keypairs'
+
+import { ValidationError } from '../common/errors'
+import { Signer } from '../models/common'
+import { Transaction } from '../models/transactions'
+import { verifyBaseTransaction } from '../models/transactions/common'
+
+import Wallet from '.'
+
+/**
+ * Uses a wallet to cryptographically sign a transaction which proves the owner of the wallet
+ * is issuing this transaction.
+ *
+ * @param wallet - A Wallet that holds your cryptographic keys.
+ * @param tx - The Transaction that is being signed.
+ * @param forMultisign - If true, changes the signature format to encode for multisigning.
+ * @returns A signed Transaction.
+ */
+function sign(wallet: Wallet, tx: Transaction, forMultisign = false): string {
+  return wallet.signTransaction(
+    tx,
+    forMultisign ? { signAs: wallet.getClassicAddress() } : { signAs: '' },
+  )
+}
+
+/**
+ * Takes several transactions (in object or blob form) and creates a single transaction with all Signers
+ * that then gets signed and returned.
+ *
+ * @param transactions - An array of Transactions (in object or blob form) to combine and sign.
+ * @returns A single signed Transaction which has all Signers from transactions within it.
+ * @throws ValidationError if:
+ * - There were no transactions given to sign
+ * - The SigningPubKey field is not the empty string in any given transaction
+ * - Any transaction is missing a Signers field.
+ */
```

Comment:
i think we could make the docstrings more clear to show that `transactions` are all already signed